lib/app/features/feed/views/pages/article_details_page/hooks/use_scroll_indicator.dart

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+final indicatorWidthProvider = StateProvider<double>((ref) {
```

Comment:
I guess it can be implemented without this provider, and the documentation says it doesn't recommend using `PlatformDispatcher.instance` ([link](https://api.flutter.dev/flutter/dart-ui/PlatformDispatcher-class.html)). Can you consider the usage of `AnimatedFractionallySizedBox` ([link](https://api.flutter.dev/flutter/widgets/AnimatedFractionallySizedBox-class.html)):

```
final scrollProgress = useState(0.05);
useScrollIndicator()
----
AnimatedFractionallySizedBox(
      alignment: Alignment.centerLeft,
      widthFactor: scrollProgress.value
)
```